### PR TITLE
Pods 7850

### PR DESCRIPTION
**Pull request by jamdade-sharad**

### CODE REVIEW

File: app/helpers/CYAChargeDHelper.scala [hunk 1]
```diff
@@ -106,4 +109,249 @@ class CYAChargeDHelper(srn: String, startDate: LocalDate, accessType: AccessType
     )
   }
 
+  def isPsprForChargeD(index: Int, isPSR: Option[Boolean]): Seq[Row] = {
```
Comment (cyeniceri) [inline app/helpers/CYAChargeDHelper.scala:117]: let's move these common methods into another class and extend it by E and D and reuse the methods

File: test/controllers/chargeD/CheckYourAnswersControllerSpec.scala [hunk 2]
```diff
@@ -21,40 +21,103 @@ import controllers.base.ControllerSpecBase
...
   ).flatten
 
-  private val jsonToPassToTemplate: JsObject = Json.obj(
-    "list" -> rows
+  private def jsonToPassToTemplate(isPSR: Boolean, isChargeInAddition: Boolean, wasAnotherPensionScheme: Boolean): JsObject = Json.obj(
+    "list" -> rows(isPSR, isChargeInAddition, wasAnotherPensionScheme)
   )
 
-  "CheckYourAnswers Controller" must {
+  "CheckYourAnswers Controller for (false, false, false)" must {
```
Comment (cyeniceri) [inline test/controllers/chargeD/CheckYourAnswersControllerSpec.scala:117]: can we give a proper name for this test ?

File: app/controllers/mccloud/IsChargeInAdditionReportedController.scala [hunk 3]
```diff
@@ -122,13 +122,26 @@ class IsChargeInAdditionReportedController @Inject()(override val messagesApi: M
...
             },
             value =>
-              for {
-                updatedAnswers <- Future.fromTry(userAnswersService.set(IsChargeInAdditionReportedPage(chargeType, index), value, mode))
-                _ <- userAnswersCacheConnector
-                  .savePartial(request.internalId, updatedAnswers.data, chargeType = Some(chargeType), memberNo = Some(index.id))
-              } yield
-                Redirect(
-                  navigator.nextPage(IsChargeInAdditionReportedPage(chargeType, index), mode, updatedAnswers, srn, startDate, accessType, version))
+              if(value) {
```
Comment (cyeniceri) [inline app/controllers/mccloud/IsChargeInAdditionReportedController.scala]: please add tests for new controllers

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: app/helpers/CYAChargeDHelper.scala (hunk 1 above)
Comment (jamdade-sharad) [reply app/helpers/CYAChargeDHelper.scala]: Created a common helper for common methods

File: test/controllers/chargeD/CheckYourAnswersControllerSpec.scala (hunk 2 above)
Comment (jamdade-sharad) [reply test/controllers/chargeD/CheckYourAnswersControllerSpec.scala]: done

File: app/controllers/mccloud/IsChargeInAdditionReportedController.scala (hunk 3 above)
Comment (jamdade-sharad) [reply app/controllers/mccloud/IsChargeInAdditionReportedController.scala]: done

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**